src/decorators/__tests__/union.decorator.test.ts

```diff
@@ -0,0 +1,26 @@
+import { IsNumber, IsString } from '../../flavors/openapi';
+import {
+    createOpenAPIDocument,
+    createOpenAPIFixtureModule,
+} from '../../testing';
+import { Union } from '../union.decorator';
+
+class Foo {
+
+    @Union([
```

Comment:
Here's a hypothetical usage of `@Union()`. 

There are a few problems here.

1. I've chosen to pass the DTO decorator *type* here rather than an instance because some/many of the instance properties do not make sense. For example, what would it mean if one of the union type members were optional/nullable and the other were not?
2. The choice to only pass in the decorator type means that we can't easily support enumerations, which is one of our main use cases for union types (e.g. `number | 'foo' | 'bar'`).
3. An alternative would be to use OpenAPI `oneOf` directly and have the caller pass in the values of `oneOf` using the appropriate schema. However, we ultimately want to build a decorator the enables `class-transformer/class-validator` as well and passing in raw input makes it harder to have a single source of truth used by all the DTO layers.
4. We could also write another input type layer that matches our usage a bit more, but this starts to be confusing for users because they'd sometime use `IsString` and sometimes use this new thing.